### PR TITLE
Signup: Do not check domain availability for WordPress.com addresses

### PR DESCRIPTION
**Pull request by bisko**
This is a fix for #5461.

On the domains step during signup, if you type an address that looks like a domain, but ends in `.wordpress.com`, a domain availability call was made, which spawned an unnecessary error message, which didn't bring anything but confusion to the user.

Now the extra call is skipped and the error message is not shown.

To test:

* Start the signup process
* Reach domain steps
* Type an address that ends in `.wordpress.com`, e.g. `nonexistingtestdomainforme.wordpress.com`
* No extra error should be shown, saying that mapping is now allowed for WordPress.com subdomains

cc @michaeldcain @coreh @lancewillett

### CODE REVIEW

**Comment by aidvu** [comment]
@klimeryk you did work on this recently. Please check if this is ok.

**Comment by klimeryk** [comment]
Thanks for this fix - it indeed corrects the confusing situation I probably introduced in https://github.com/Automattic/wp-calypso/pull/5331. However, I'd love to have the warning show up during search in Domain Management. I can imagine confused users trying to add a new `*.wordpress.com` site that way and this notice should help to steer them the right way.

**Comment by umurkontaci** [comment]
Code looks good, tested and works OK on signup, has a slight regression on Domain Management as @klimeryk pointed out.

If you search for `*.wordpress.com` on Domain Management, you get:
image

**Comment by bisko** [comment]
I updated the PR to disable the domain availability check only if it is included in the signup check. The Domain management screen is behaving like it should.

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -274,7 +274,15 @@ const RegisterDomainStep = React.createClass( {
 		async.parallel(
 			[
 				callback => {
-					if ( ! domain.match( /.{3,}\..{2,}/ ) ) {
+					if ( !domain.match( /.{3,}\..{2,}/ ) ) {
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: There's a space now missing after `!` :)

File: client/components/domains/register-domain-step/index.jsx [hunk 2]
```diff
@@ -274,7 +274,15 @@ const RegisterDomainStep = React.createClass( {
...
+						return callback();
+					}
+
+					/**
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: Not sure this comment is really needed - I think the condition `this.props.isSignupStep && domain.match( /\.wordpress\.com$/ )` is pretty much self-explanatory.

**Comment by klimeryk** [comment]
Thanks for the fix, works as advertised! 👍 

I've left some nitpicks - feel free to ignore/discuss the comments you don't agree with :)
Additionally, I think you should add the `isSignupStep` to `RegisterDomainStep`'s `propTypes`.

File: client/components/domains/register-domain-step/index.jsx (hunk 1 above)
Comment (bisko) [inline client/components/domains/register-domain-step/index.jsx]: Oops. IDE misconfiguration can be bad sometimes. Fixed and fixed the linter to tell me about these.

File: client/components/domains/register-domain-step/index.jsx (hunk 2 above)
Comment (bisko) [inline client/components/domains/register-domain-step/index.jsx]: Good suggestion. I removed the comment.

**Comment by bisko** [comment]
> Additionally, I think you should add the `isSignupStep` to `RegisterDomainStep`'s `propTypes`.

I definitely missed that step. I added `isSignupStep` to the `propTypes`

**Comment by klimeryk** [comment]
Thanks! LGTM! 👍

**Pull request merged**